### PR TITLE
DRAFT: DDD introduction

### PR DESCRIPTION
**Pull request by dieterve**
Initial draft, still needs work.

@yannickdepauw: I would like to add something visual to this post to make it more attractive (instead of a blob of text. Maybe you (or anyone else) has an idea for a fitting image?

### CODE REVIEW

**Comment by yannickdepauw** [comment]
All up for helping out. Any examples of things you have in mind? What illustrations could it be?

File: _posts/2016-07-01-introduction-to-domain-drive-design.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
+---
+layout: post
+title: "Introduction to domain driven design"
+date: 2016-07-01
+authors: ["dieterve", "bram"]
+categories: ["ddd", "domain driven design", "cqrs"]
+---
+
+As applications grow bigger and bigger, our codebase needs to accommodate for more and more business actions. After a while you will have code that impacts a large amount of business actions without knowing exactly what they are. Changes to that kind become more and more stressful. We had long been feeling the need to handle this complexity of our applications in a more explicit and clearer way.
```
Comment (jdrieghe) [inline _posts/2016-07-01-introduction-to-domain-drive-design.md]: `... "our codebase" ... "you will have" ...`

Best to choose one point of view and remain consistent in using it

File: _posts/2016-07-01-introduction-to-domain-drive-design.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+## What is Domain Driven Design?
+
+Something about defining business before starting, about finding loops in the business, possible problems and pitfalls. Something about documenting and sharing the business with the team.
+
+Domain Driven Design aims to help you define the rules and actions of the business.
+
+It is not the holy grail to software development. If your application is simply about adding and managing records then a DDD approach is not as suitable and might even add more overhead.
+
+If your software involves multiple departments in an organization, various types of products or large amounts of data then complexity is bound to rear it’s ugly head.
```
Comment (jdrieghe) [inline _posts/2016-07-01-introduction-to-domain-drive-design.md]: it's => its

**Comment by dieterve** [comment]
> All up for helping out. Any examples of things you have in mind? What illustrations could it be?

@yannickdepauw: I don't have much inspiration. The main goal of domain drive design is to bring clarity in complex business logic, to be more explicit about actions/decisions/restrictions. Maybe something with a person clearly seeing paths. Or a person standing in front of several hiking trails and seeing multiple arrows which say "Add product to basket", "Add product to wishlist", ... (this is what happens when I get creative).

@bramdevries and @jdrieghe might have some analogies as well.

**Comment by jdrieghe** [comment]
How about something with overlapping Venn diagrams for "Language", "Business" and "Software" ?

**Comment by batusa** [comment]
@jdrieghe I think it is good. business and software has to 'talk' the same language.

**Comment by dieterve** [comment]
Did some changes, for me its oke now. @bramdevries can you also check again?

@yannickdepauw: any ideas for an image?

**Comment by bramdevries** [comment]
Looks good to me

**Comment by hannesvdvreken** [comment]
would be nice to have a nice looking image for each future blog post like digital ocean does:  https://www.digitalocean.com/company/blog/

**Comment by yannickdepauw** [comment]
@dieterve try to throw this online today, ok? If you want me to look further into pics, just holla!

**Pull request merged**